addons/website_sale_coupon/models/sale_order.py

```diff
@@ -8,6 +8,25 @@
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
+    def _try_pending_coupon(self):
+        if not request:
+            return
+
+        status = {key: value for key, value in request.session.get('pending_coupon', {}).items() if key == 'promo_code'}
+
+        if self and 'promo_code' in status:
+            coupon_code = status.get('promo_code')
+            status.update(self.env['sale.coupon.apply.code'].apply_coupon(self, coupon_code))
```

Comment:
apply_coupon is probably th emost important of this function _try_pending_coupon... give him a specific line ;)